scripts/o3de/tests/test_engine_properties.py

```diff
@@ -64,9 +92,18 @@ def save_o3de_manifest(new_engine_data: dict, engine_path: pathlib.Path) -> bool
         with patch('o3de.manifest.get_engine_json_data', side_effect=get_engine_json_data) as get_engine_json_data_patch, \
                 patch('o3de.manifest.save_o3de_manifest', side_effect=save_o3de_manifest) as save_o3de_manifest_patch, \
                 patch('o3de.manifest.get_registered', side_effect=get_engine_path) as get_registered_patch:
-            result = engine_properties.edit_engine_props(engine_path, engine_name, engine_new_name, engine_version)
+            result = engine_properties.edit_engine_props(engine_path, engine_name, engine_new_name, engine_version,
+                engine_display_version, add_gem_names, delete_gem_names, replace_gem_names,
+                add_api_versions, delete_api_versions, replace_api_versions)
             assert result == expected_result
-            if engine_new_name:
-                assert self.engine_json.data.get('engine_name', '') == engine_new_name
-            if engine_version:
-                assert self.engine_json.data.get('O3DEVersion', '') == engine_version
+            if result == 0:
+                if engine_new_name:
+                    assert self.engine_json.data.get('engine_name', '') == engine_new_name
+                if engine_version:
+                    assert self.engine_json.data.get('O3DEVersion', '') == engine_version
```

Comment:
Do we still need to check the O3DEVersion in test since it is deprecated?

Reply:
I left it in for now, but will deprecate (with a warning first) then remove it when we add version info to the main `engine.json` and core `gem.json` files.